### PR TITLE
CLN: Delete no-op check

### PR DESCRIPTION
**Pull request by dsaxton**
Maybe the note is somehow useful, otherwise I think this can be cleaned?

### CODE REVIEW

File: pandas/core/arrays/categorical.py [hunk 1]
```diff
@@ -2650,11 +2650,8 @@ def _convert_to_list_like(list_like):
         return list_like
     if is_sequence(list_like) or isinstance(list_like, tuple) or is_iterator(list_like):
         return list(list_like)
-    elif is_scalar(list_like):
-        return [list_like]
-    else:
-        # TODO: is this reached?
-        return [list_like]
+
+    return [list_like]
```
Comment (jbrockmendel) [inline pandas/core/arrays/categorical.py:2654]: any idea why we're not doing `if is_list_like(x): return x; return [x]`?

Comment (dsaxton) [reply pandas/core/arrays/categorical.py:2654]: Not sure, that's cleaner and will try

Comment (dsaxton) [reply pandas/core/arrays/categorical.py:2654]: Looks like a lot of things fail if we try that (e.g., tuples need to be converted to lists I think)

**Comment by jreback** [comment]
ok for now..

can you open an issue / PR on moving this routine elsewhere (maybe pandas.core.common). We do this in quite a few places and would be nice to centalize.

**Pull request merged**